spec/models/pacbio/library_factory_spec.rb

```diff
@@ -42,6 +44,14 @@
       expect(factory).to_not be_valid
       expect(factory.errors.full_messages).to_not be_empty
     end
+
+    it 'produces an error if the request contains an empty cost_code' do
+      library_attributes << attributes_for(:pacbio_library).merge(requests: [{id: request_empty_cost_code.id, type: 'requests'}])
+      factory = Pacbio::LibraryFactory.new(library_attributes)
+      expect(factory).to_not be_valid
+      expect(factory.errors.full_messages).to eq(['Cost code must be present'])
+    end
```

Comment:
Hi @BenTopping , Harry and I were looking at this together and we reckon it would be nicer to call 'save' explicitly before the expectations. We realised that the only reason it's calling the validation in here is because the 'be_valid' is calling it, which seems a bit crazy and magical. Does that make sense? Thanks